code/front/js/productFront.js

```diff
@@ -0,0 +1,148 @@
+// already have in the console
+
+/*
+** create Product View
+*/
+
+
+//populate data, image ,description, price using keys/values
+function createIndividualProductView(data) {
+    //Dom access to existing ids + classes
+    let individualProductName = document.getElementById('title');
+    let individualProductPrice = document.getElementById('price');
+    let individualProductDescription = document.getElementById('description');
+    
+    let imgParent = document.getElementById('item__img');
+
+    // create element 
+    let individualProductImg = document.createElement('img');
+
+    //populate 
+    individualProductName.innerText = data.name;
+    individualProductPrice.innerText = data.price;
+    individualProductDescription.innerText = data.description;
+    individualProductImg.src = data.imageUrl;
+
+    //append
+    imgParent.appendChild(individualProductImg);
+
+    return createIndividualProductView;
+}
+
+/*
+** Populate Dropdown
+*/
+
+/*function for populating the dropdown logic - decomposition into smallest parts
+I have the array I want to access nested within the object 
+Access the array
+iterate over the array
+Access the dom 
+populate the dom 
+append to dom
+*/
+
+const populateDropdown = (data) => {
+    //access select node
+    let select = document.getElementById('colorsParent');
+    //Access the array
+    const colors = data.colors;
+    //iterate over the array
+    for(let i = 0; i < colors.length; i++) {
+        //create elements 
+        let dropdownSelector = document.createElement('option');
+        //populate the selector
+        dropdownSelector.innerText = data.colors[i]; //needed [i] otherwise options would be on one line
+        //append
+        select.appendChild(dropdownSelector);
+    }
+}
+
+/*
+** | from openClassRooms 
+*/
+
+/* cart can be made as an array containing three things
+- product id, quanitity, color
+use localStroage to access this array
+when you add a product to the cart, add a new element to the array
+when you add the exact same product, increase the quantity 
+*/
+
+var button = document.getElementById('addToCart');  //console.log(button); working
+button.addEventListener('click', () => { 
+  var option = document.getElementById('colorsParent').value;
+  //if color is unselected alert user
+  if (option == '') {
+    alert('please select an item color choice from the dropdown menu');
+    return;
+  }
+  //if value of selector is 0, throw error message
+  var quantity = document.getElementById('quantity').value; 
+  var interestedProduct = {_id: product._id, color: option, quantity: parseInt(quantity), name: product.name, description: product.description, image: product.imageUrl, price: product.price, altText: product.altText};
+  if (quantity == 0) {
+    alert('please select a value');
+    return; 
+  }
+  console.log(interestedProduct); //array
+  //call function
+  addToShoppingCart(interestedProduct); 
+});
+
+/*
+** | add to shopping cart
```

Comment:
add more meaningful comments in all the functions:
what does the function receive?
what does the function do?
what does the function return?